### PR TITLE
Refactoring SpiLogger to better support conditional logging

### PR DESCRIPTION
**Pull request by rbygrave**
Initial step, remove trace() logging from SpiLogger
- Not bothered to log when changing isolation level
- Not bothering to log at end of query only transaction

### CODE REVIEW

File: ebean-core/src/main/java/io/ebeaninternal/server/core/PersistRequestBean.java [hunk 1]
```diff
@@ -932,16 +932,16 @@ private void logSummaryMessage() {
     String name = beanDescriptor.name();
     switch (type) {
       case INSERT:
-        transaction.logSummary("Inserted [" + name + "] [" + (idValue == null ? "" : idValue) + draft);
+        transaction.logSummary("Inserted [" , name , "] [" , (idValue == null ? "" : idValue.toString()) , draft);
```
Comment (rPraml) [inline ebean-core/src/main/java/io/ebeaninternal/server/core/PersistRequestBean.java:935]: Maybe change logger signature to `logSummary(Object... msg)`

File: ebean-core/src/main/java/io/ebeaninternal/server/core/PersistRequestCallableSql.java [hunk 2]
```diff
@@ -86,7 +82,7 @@ public void postExecute() {
       persistExecute.collectSqlCall(label, startNanos);
     }
     if (transaction.isLogSummary()) {
-      transaction.logSummary("CallableSql label[" + callableSql.getLabel() + "]" + " rows[" + rowCount + "]" + " bind[" + bindLog + "]");
+      transaction.logSummary("CallableSql label[", callableSql.getLabel(), "] rows[", String.valueOf(rowCount), "] bind[", bindLog, "]");
```
Comment (rPraml) [inline ebean-core/src/main/java/io/ebeaninternal/server/core/PersistRequestCallableSql.java:85]: +1 for `logSummary(Object... msg)`

**Review by rPraml — COMMENTED** [review]
looks good so far, looking forward to trying it out

File: ebean-core/src/main/java/io/ebeaninternal/server/core/PersistRequestBean.java (hunk 1 above)
Comment (rbygrave) [reply ebean-core/src/main/java/io/ebeaninternal/server/core/PersistRequestBean.java:935]: Yes agreed. It's starting to tell me it _could even_ be `logSummary(String msgFormat, Object... args)` ...

**Pull request merged**